pkg/apis/admissionregistration/validation/validation_test.go

```diff
@@ -2645,6 +2645,46 @@ func TestValidateValidatingAdmissionPolicy(t *testing.T) {
 			},
 		},
 		expectedError: `spec.validations[0].messageExpression: Invalid value: "object.x in [1, 2, ": compilation failed: ERROR: <input>:1:20: Syntax error: missing ']' at '<EOF>`,
+	}, {
+		name: "messageExpression may reference authorizer",
+		config: &admissionregistration.ValidatingAdmissionPolicy{
```

Comment:
I believe this is a test only of validation. Could you add tests that uses the authorizer value in a message expression? If its possible to show that the caching is working across the rule expressions and the message expressions that would also be good to see

Reply:
Thanks for catching this. I mistakenly thought I had included integration test coverage too. Added a new test case in my last push. I'll add an expression/messageExpression caching test once the caching PR merges.